azkaban-exec-server/src/test/java/azkaban/execapp/FlowRunnerTest2.java

```diff
@@ -648,6 +643,8 @@ public void testRetryOnFailure() throws Exception {
       System.out.println(inNode.getId() + " > " + inNode.getStatus());
     }
 
+    assertStatus("jobb:innerFlow", Status.SKIPPED);
+    InteractiveTestJob.clearTestJobs("jobb:innerJobB", "jobb:innerJobC");
```

Comment:
When a job run is started, the created job type instance gets added into `InteractiveTestJob`'s map. If a job is retried, another instance is created, but for `InteractiveTestJob` the key=jobname is the same.

My commit message says:
> Clear previous test jobs so that .succeedJob() is called on the retried job instance.

So occasionally it happened that `InteractiveTestJob.getTestJob("jobb:innerJobB").succeedJob` was targeted at the original failed instance. After that the retry attempt instance was added into `InteractiveTestJob`'s map, but `succeedJob` was never called on it, so it was left in `RUNNING` state.